### ISSUE TITLE
Custom domain

### ISSUE THREAD

**Issue by simonwfarrow** [architect/arc.codes#147]
Hi, 
I'm new to arch from v6, it has made it very easy to configure AWS for my functions, DB etc. I would now like to add my own domain to handle the http requests and serve my static site. I've come across posts referring to the @domain option and the dns command , however this appears to have been removed. How can I map my custom domain (registered with route 53) so it is used on each arc deploy rather than the generated domain?  Do I need to do this manually? If so, would this be overwritten on a new deploy?
What are the plans for this functionality - the documents suggest it is being re-visited.

**Comment by brianleroux** [comment]
Hey @simonwfarrow thx for writing us! Def an open issue to solve. I'll document the manual steps below.

0.) Ensure you have a `@cdn` pragma. While not strictly necessary I would consider is a best practice. This will generate a CloudFront CDN's for your API Gateway and S3 buckets when you run `arc deploy`. ⚠️ If you remove `@cdn` the next deploy will remove them! 

1.) Architect has concepts of `staging` and `production` by default. You deploy to staging by running `arc deploy` and you deploy to production running `arc deploy production`. under the hood these create CloudFormation stacks `AppnameStaging` and `AppnameProduction` respectively. You'll want to decide what the domains for each of these are. 

2.) You need a certificate to serve your traffic over HTTPS. This is accomplished by opening up the AWS console and navigating to AWS Certificate Manager (ACM) in `us-east-1` and creating a certificate. ⚠️ This MUST be done in `us-east-1`. Creating a cert request will generate CNAME records which ACM uses to verify you actually own the domain. Note these values down.

3.) Open up Route53 in the AWS console. If you are not using Route53 then go to whomever is providing you with DNS and enter the CNAME records you got from Step 2. Go back to ACM to verify the certificate has been issued. This can take up to 10 minutes but usually happens quickly.

4.) Open up CloudFront in the console and find the distributions for your staging and production API Gateway origins. Select the staging distribution and click 'Distribution Settings' -> 'Edit'. Add the CNAME for the domain you desire in 'Alternate Domain Names' text input and then select 'Custom SSL Certificate (example.com):' with the value of the certiciate you created above. Save the settings by clicking 'Yes, Edit' in the bottom right and repeat for the production domain. 

Take note of the CloudFront domains for staging and production. They look like: `xxx.cloudfront.net`.

5.) Finally! Go back to Route53 (or your DNS provider) and create CNAME (or Alias) records pointing to the `xxx..cloudfront.net` domains you noted above. Wait about 5-15 minutes and your new domains should be live. 

---

We want to automate this but have encountered a lot of resistence to using Route53 and CloudFront. Many devs want to use a third party DNS provider and third party CDN provider so we decided to 'wait and see' how the response to this release went. Sorry it is so manual! At least you only have to do this once. I'm going to leave this open because we should add docs that outline above and have screenshots.

**Comment by simonwfarrow** [comment]
Many thanks @brianleroux for the speedy and detailed reply.
I will try this out and revert back with any findings.

**Comment by simonwfarrow** [comment]
That worked perfectly. Couple of minor things to point out, mostly due to my lack of understanding I am sure, but may help in the guide.

For step 0, I added the @cdn option, this generated the CloudFront distribution  - this caused caching on all my GET requests ; following your user guide I had to add the Cache-Control headers and then invalidate the cache on CloudFront by creating a new invalidation for the new distribution.

For step 4, ```Custom SSL Certificate (example.com):' with the value of the certiciate you created above```  - I hadn't realised this was the ARN of the certificate.

For step 5, creating the CNAME involved selecting my Hosted Zone, then create Record Set of type CNAME and adding the CloudFront domain in there. 

Thanks again for your guidance. Loving arc!
